c_api/tests/c_api_behavior_test.cpp

```diff
@@ -113,3 +113,453 @@ TEST_F(CapiTest, TestBehaviorGetRuntimeCapabilities) {
 
   inner(TI_ARCH_VULKAN);
 }
+
+TEST_F(CapiTest, TestBehaviorAllocateMemory) {
+  TiError error = TI_ERROR_SUCCESS;
+  TiMemoryAllocateInfo allocate_info;
+  allocate_info.size = 1024;
+  auto inner = [&](TiArch arch) {
+    if (ti::is_arch_available(arch)) {
+      // Attempt to allocate memory with size of 1024
+      TiRuntime runtime = ti_create_runtime(arch);
+      for (int i = 0; i < 4; ++i) {
+        allocate_info.usage = TI_MEMORY_USAGE_STORAGE_BIT << i;
+        TiMemory memory = ti_allocate_memory(runtime, &allocate_info);
+        TI_ASSERT(memory != TI_NULL_HANDLE);
+      }
+
+      // Attempt to run out of the memory
+      {
+        allocate_info.size = 1000000000000000000;
```

Comment:
Don't reuse `allocate_info`, it's hard to understand the process when it's constantly changing. Fill it up in each test block, or just make a utility filler function instead.